### PR TITLE
Clean up some move related code and matches.

### PR DESCRIPTION
**Pull request by asvitkine**
## Change Summary & Additional Notes

- Remove some unnecessary arrived.isEmpty() checks in a branch where that's already been checked.
- Rename some matchers to make more sense and streamline their implementations.
- Remove unnecessary params.

### CODE REVIEW

File: game-app/game-core/src/main/java/games/strategy/triplea/delegate/Matches.java [hunk 1]
```diff
@@ -1281,21 +1287,24 @@ public static Predicate<Territory> isTerritoryEnemyAndNotUnownedWater(
...
+    final RelationshipTracker relationshipTracker = player.getData().getRelationshipTracker();
+    return territoryNotImpassibleOrRestrictedOrNeutralWaterAndNotOwnedBy(player)
+        .and(t -> relationshipTracker.canTakeOverOwnedTerritory(player, t.getOwner()));
+  }
+
+  private static Predicate<Territory> territoryNotImpassibleOrRestrictedOrNeutralWaterAndNotOwnedBy(
+      final GamePlayer player) {
+    return not(isTerritoryOwnedBy(player))
+        .and(not(territoryIsUnownedWater()))
+        .and(territoryIsPassableAndNotRestricted(player, player.getData().getProperties()));
```
Comment (RoiEXLab) [inline game-app/game-core/src/main/java/games/strategy/triplea/delegate/Matches.java:1307]: If you're simplifying the method signature anyways, would it be a benefit to have `territoryIsPassableAndNotRestricted` only accept a `GamePlayer` and derive the properties from that? Or is this just a special case that happens only here?

**Review by RoiEXLab — APPROVED** [review]

File: game-app/game-core/src/main/java/games/strategy/triplea/delegate/Matches.java (hunk 1 above)
Comment (asvitkine) [reply game-app/game-core/src/main/java/games/strategy/triplea/delegate/Matches.java:1307]: One danger is if the null player instance is ever passed, that one will not have a null data. I filed an issue about getting rid of a global null player instance and instead having a null instance associated with the game data, but that's a bigger refactoring.

So it's not necessarily safe to do - depends on call sites.

File: game-app/game-core/src/main/java/games/strategy/triplea/delegate/MovePerformer.java [hunk 2]
```diff
@@ -256,9 +254,7 @@ public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
               // Ignore Trn on Trn forces.
               if (Properties.getIgnoreTransportInMovement(bridge.getData().getProperties())) {
                 final boolean allOwnedTransports =
-                    !arrived.isEmpty()
-                        && arrived.stream()
-                            .allMatch(Matches.unitIsTransportButNotCombatTransport());
```
[the remark this answers is not in the document]
Comment (asvitkine) [reply game-app/game-core/src/main/java/games/strategy/triplea/delegate/MovePerformer.java:261]: Ack.

File: game-app/game-core/src/main/java/games/strategy/triplea/delegate/Matches.java (hunk 1 above)
Comment (asvitkine) [reply game-app/game-core/src/main/java/games/strategy/triplea/delegate/Matches.java:1307]: Keeping as-is per the above. Once we can change the engine such that the null player is guaranteed to have a non-null data, I'd be more comfortable refactoring broader in this way.

**Pull request merged**